snuba/consumers/utils.py

```diff
@@ -0,0 +1,52 @@
+import time
+from logging import Logger
+
+from confluent_kafka import KafkaException
+from confluent_kafka.admin import AdminClient
+
+from snuba.utils.streams.configuration_builder import get_default_kafka_configuration
+from snuba.utils.streams.topics import Topic
+
+
+def get_partition_count_from_kafka_admin(topic: Topic, logger: Logger) -> int:
```

Comment:
```suggestion
def get_partition_count(topic: Topic) -> int:
```

I think `get_partition_count` is a better name. The fact that we are using an admin client is an implementation detail that I don't think the caller of this function should worry about.

Why do you need to pass the logger into this function?

Reply:
cool yeah I can simplify the name. 

> Why do you need to pass the logger into this function?

I figured this would retain the logger instance associated with the specific consumer if this gets used by more than the executor consumers so we'd know if it was the executor, scheduler etc. Maybe that's not the best way to do that?

Reply:
We don't typically use the logger name for that. Otherwise we would need pass the logger around to every function and that seems like a pain. Sentry always tags the server_name anyway. And if you were looking at the raw server logs well, I feel like you should probably know which server you are looking at anyway.